website/content/docs/configuration/seal/kmip.mdx

```diff
@@ -0,0 +1,161 @@
+---
+sidebar_label: KMIP Unseal
+description: |-
+  The KMIP seal configures OpenBao to use the KMIP protocol as the
+  autounseal mechanism.
+---
+# `kmip` seal
+
+The KMIP seal configures OpenBao to use a KMS or an HSM through the KMIP protocol as the autounseal
+mechanism. OpenBao's KMIP support is activated by one of the following:
+
+- The presence of a `seal "kmip"` block in OpenBao's configuration file
+- The presence of the environment variable `BAO_SEAL_TYPE` set to `kmip`. If enabling via environment
+  variable, all other required values (i.e. `BAO_KMIP_ENDPOINT`) must be also
+  supplied.
+
+:::info
+
+OpenBao requires key material to be created externally
+before initializing the instance. OpenBao will not attempt to create its own
+key material.
+
+:::
+
+:::warning
+
+Because KMIP auto-unseal was not present in Vault v1.14 OSS, it is not
+expected that it is [seal-compatible](/docs/policies/migration/); manual
+migration of data between nodes may be required.
+
+:::
+
+## KMIP Server Requirements
+
+- Must support one of the following KMIP protocol version: v1.2, v1.3 or v1.4.
+- Must support Encrypt & Decrypt operations
+- Must support AES symmetric keys or RSA key pairs
+- Must support one of the following encryption algorithms
+  - AES_GCM
+  - RSA_OAEP_SHA256
+  - RSA_OAEP_SHA384
+  - RSA_OAEP_SHA512
+
+## `kmip` example
+
+This example shows configuring KMIP seal through the OpenBao configuration
+file by providing all the required values:
+
+```hcl
+seal "kmip" {
+  endpoint = "eu-west-rbx.okms.ovh.net:5696"
+  kms_key_id = "c1afd090-e8ae-4d03-a81f-3938b7e2be02"
+  client_cert = "client-cert.pem"
+  client_key = "client-key.pem"
+
+  // Optional parameters:
+  //
+  // ca_cert = ""
+  // server_name = ""
+  // timeout = 10
+  // encrypt_alg = ""
+  // tls12_ciphers = ""
+}
+```
+
+To create the key in the KMIP Server, it is recommended to use your KMIP server vendor tooling.
+
+Example for creating a KMIP key on OVHcloud's KMS offer using the [okms command line tool](https://github.com/ovh/okms-cli):
+
+```bash
+$ okms kmip create symmetric --alg AES --size 256 --usage encrypt,decrypt --name bao_seal_key
```

Comment:
If I understand the tool correctly, this should work on any KMIP server assuming correct permissions, yes? My understanding of the README is that it uses KMIP to establish parameters, and thus should work fine?

Reply:
@cipherboy your understanding is correct. While the tool is specifically developped for OVH KMS, the `kmip` subcommands only use the kmip protocol and would work on any KMIP enabled KMS, not only OVH. It uses mTLS for authentication, and has a dedicted config file section. At some point we may move the kmip subcommand into a standalone kmip tool.

Other commands in the tool will however use the OVH rest api, and thus whill only work against OVH kms.

Reply:
Alternatively a tool of mine that can be used is https://github.com/phsym/kmip-explorer. I may move it under ovh org some day